Add documentation for SCD_RUN_BALER env variable for Magento Cloud (#6895)
magento/devdocs: src/cloud/env/variables-build.md

-  `skip_di_clearing`
-  `skip_di_compilation`

### `SCD_USE_BALER`

-  **Default**—_Not set_
-  **Version**—Magento 2.3.0 and later

[Baler](https://github.com/magento/baler) scans your generated JavaScript code and creates an optimized JavaScript bundle. Deploying the optimized bundle to your site can reduce the number of network requests when loading your site and improve page load times.

Set to `true` to run Baler after performing static content deployment.

```yaml
stage:
  build:
    SCD_USE_BALER: true
```

{:.bs-callout-info}

> You must install and configure the Baler module before using this feature. Because Baler is currently in alpha release, enable this option only on Staging environments.

Reviewer: Just to make it clearer that you can try it in integration as well, but also to I think it's better to say we _recommend_ as it's optional.
```suggestion
You must install and configure the Baler module before using this feature. Because Baler is currently in alpha release, we do not recommend using it in Production environments.
```
